model/src/main/java/org/projectnessie/api/NamespaceApi.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.api;
+
+import javax.validation.constraints.NotNull;
+import org.projectnessie.api.params.NamespaceParams;
+import org.projectnessie.api.params.NamespacesParams;
+import org.projectnessie.error.NessieNamespaceAlreadyExistsException;
+import org.projectnessie.error.NessieNamespaceNotEmptyException;
+import org.projectnessie.error.NessieNamespaceNotFoundException;
+import org.projectnessie.error.NessieReferenceNotFoundException;
+import org.projectnessie.model.GetNamespacesResponse;
+import org.projectnessie.model.Namespace;
+
+public interface NamespaceApi {
+
+  /**
+   * Creates a new namespace.
+   *
+   * @param params The {@link NamespaceParams} that includes the parameters for the API call.
+   * @return A {@link Namespace} instance if creating the namespace succeeded.
+   * @throws NessieReferenceNotFoundException If the reference could not be found.
+   * @throws NessieNamespaceAlreadyExistsException If the namespace already exists.
+   */
+  Namespace createNamespace(@NotNull NamespaceParams params)
+      throws NessieNamespaceAlreadyExistsException, NessieReferenceNotFoundException;
```

Comment:
Since this operation makes a commit, I though we ought to return a `Branch` with the up-to-date reference info... same as what `TreeApi.commitMultipleOperations` returns.

Does returning `Namespace` to the caller actually communicates new information that the caller does not have already?

Also, since a commit is made I think we have to declare `NessieConflictException`. Even though it is not technically possible now, it may be possible later when namespace have properties.

Reply:
returning a `Branch` seems a bit non-expected from a user's perspective. Returning a `Namespace` does not carry any new information currently, but it will potentially in the future